vortex-array/src/compute/compare.rs

```diff
@@ -89,12 +89,31 @@ pub fn compare(left: &Array, right: &Array, operator: Operator) -> VortexResult<
         vortex_bail!("Compare operations only support arrays of the same type");
     }
 
-    if ConstantArray::try_from(left).is_ok() {
-        let scalar = scalar_at(left, 0)?;
-        let left_const = ConstantArray::new(scalar, left.len()).into_array();
-        return compare(right, &left_const, operator.swap());
+    use crate::stats::ArrayStatistics;
+    use crate::stats::Stat;
+
+    let l_is_const = ConstantArray::try_from(left).is_ok()|| left.statistics().get_as::<bool>(Stat::IsConstant).unwrap_or_default();
+    let r_is_const = ConstantArray::try_from(right).is_ok()|| right.statistics().get_as::<bool>(Stat::IsConstant).unwrap_or_default();
```

Comment:
might be easier to compare encoding ids?

Reply:
That's what I did a few PRs ago and someone steered me towards `try_from`, but I agree